pkg/api/customization/cluster/actions_cis.go

```diff
@@ -73,16 +73,51 @@ func (a ActionHandler) runCisScan(actionName string, action *types.Action, apiCo
 			fmt.Sprintf("CIS scan already running on cluster"))
 	}
 
-	if cisScanConfig.OverrideBenchmarkVersion != "" {
-		_, err := a.CisBenchmarkVersionLister.Get(namespace.GlobalNamespace, cisScanConfig.OverrideBenchmarkVersion)
-		if err != nil {
-			if kerrors.IsNotFound(err) {
+	validBenchmarkVersions, err := a.getValidBenchmarkVersions()
+	if err != nil {
+		logrus.Errorf("error fetching benchmark versions: %v", err)
+		return httperror.WrapAPIError(err, httperror.ServerError, "error fetching cis benchmark versions %v")
+	}
+
+	if cisScanConfig.OverrideBenchmarkVersion != "" &&
+		!validBenchmarkVersions[cisScanConfig.OverrideBenchmarkVersion] {
+		return httperror.WrapAPIError(err, httperror.InvalidOption,
+			fmt.Sprintf("invalid override benchmark version specified"))
+	}
+
+	userContext, err := a.ClusterManager.UserContext(apiContext.ID)
+	if err != nil {
+		logrus.Errorf("error fetching user context for cluster %v: %v", apiContext.ID, err)
+		return httperror.WrapAPIError(err, httperror.ServerError,
+			fmt.Sprintf("error fetching user contect for cluster %v", apiContext.ID))
+	}
+
+	cm, err := userContext.Core.ConfigMaps(v3.DefaultNamespaceForCis).Controller().Lister().Get(v3.DefaultNamespaceForCis, v3.ConfigMapNameForUserConfig)
```

Comment:
I don't think that getting a lister from a userContext will work properly and consistently in an HA setup. I don't think you'll have access to the cache on all api server nodes.

Reply:
Thanks for the deep insight, I didn't know about this. I have changed it to Get instead of Lister.